frontend/src/components/data-table/column-formatting/applyFormat.ts

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import { DataType } from "@/core/kernel/messages";
+import { type FormatOption } from "./types";
+import { prettyNumber, prettyScientificNumber } from "@/utils/numbers";
+
+// Apply formatting to a value given a format and data type
+export const applyFormat = (
```

Comment:
it would be great to add a few unit tests for `applyFormat`. we use https://vitest.dev/ for unit testing. there should be many examples in the repo

Reply:
I was planned to write unit tests after the implementations are ok

Reply:
that sounds good - i think that makes sense, to write these unit tests after since this may change